llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -198,6 +199,16 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST) {
 
   getActionDefinitionsBuilder(G_FRAME_INDEX).legalFor({p0});
 
+  // FP Operations
+
+  if (ST.hasStdExtF()) {
```

Comment:
X86 has also a fleet of extensions, but RISC-V will probably win. They do it slightly different:
https://github.com/llvm/llvm-project/blob/af447dde5e723e925b9c8fdc6715b07f0caf6161/llvm/lib/Target/X86/GISel/X86LegalizerInfo.cpp#L420